src/libexpr/attr-path.cc

```diff
@@ -36,18 +36,18 @@ std::vector<Symbol> parseAttrPath(EvalState & state, std::string_view s)
 {
     std::vector<Symbol> res;
     for (auto & a : parseAttrPath(s))
-        res.push_back(state.symbols.create(a));
+        res.emplace_back(a);
```

Comment:
I'm not sure I understand this. How does this guarantee that the symbol is unique (i.e. pointer-comparable)?

Reply:
it does not, as of this PR `Symbol` does not confer a uniqueness guarantee, only a "this is a nix identifier" semantics. we can remove `Symbol` and rename `SymbolIdx -> Symbol` to restore that guarantee to the name if that's preferrable